modules/bpi/bpi.images.inc

```diff
@@ -213,3 +229,71 @@ function bpi_fetch_image($filename, $image_url) {
 
   return file_save_data($result->data, $filename, FILE_EXISTS_REPLACE);
 }
+
+/**
+ * Compute the image type from a field name on a node.
```

Comment:
The bpi web service supporting assets (rather than just images) uses a custom string, the asset type, to distinguish between different types of assets connected to content (cf. https://docs.google.com/document/d/1IAUoEo1MgdVkQCh5sGPhSNLuw5Wau1Gg8UsmbVErMUE/edit#heading=h.herx6gykkgqz). A piece of content can have multiple assets of each type. Inline images (in the body field) are stored with the type "body".